src/modm/architecture/interface/can_message.hpp.in

```diff
@@ -21,22 +22,21 @@
 namespace modm::can
 {
 
-/// Representation of a CAN message
 /// @ingroup modm_architecture_can
 struct Message
 {
 	inline Message(uint32_t inIdentifier = 0, uint8_t inLength = 0) :
-		identifier(inIdentifier), flags(), length(inLength)
+		identifier(inIdentifier), flags()
 	{
+        setLength(inLength);
 	}
 
-	// Create CAN message from long data in Network Order.
-	inline Message(uint32_t inIdentifier, uint8_t inLength, const uint64_t &inData, bool extended=false) :
-		identifier(inIdentifier), length(std::min(inLength, uint8_t(8)))
+	inline Message(uint32_t inIdentifier, uint8_t inLength, const uint64_t &inData, bool extended = false) :
+		Message(inIdentifier, inLength)
 	{
 		flags.extended = extended;
 		const uint8_t *inDataB = reinterpret_cast<const uint8_t *>(&inData);
-		for (uint8_t ii = 0; ii < length; ++ii)
+		for (uint8_t ii = 0; ii < dlcConversionTable[length]; ++ii)
```

Comment:
Is this meant to handle sizes greater than 8? This change seems a bit odd. `dlcConversionTable[length]` is equal to `length` for all that are valid length values. `length > 8` would always cause out-of-bounds memory accesses.

Reply:
I have resolved this by reverting the change to the previous version